### PR TITLE
Initial

### PR DESCRIPTION
**Pull request by Pugazhenthiran-freshworks**

### CODE REVIEW

File: app/components/todos-list.hbs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    {{#each this.pendingTodos as |todo|}}
+    <li data-id={{todo.id}} data-status={{todo.status}}>
+        {{!-- <span>{{todo.title}}</span> --}}
+        <input class="todoTitleInput" aria-label={{todo.title}} {{on "blur" (fn this.onEditTaskTitle todo.id)}}
+            value={{todo.title}} />
+        <button class="inProgressBtn" type="button" {{on "click" (fn this.updateStatus todo.id "inProgress"
+            )}}>Start</button>
+        <button class="completeBtn" type="button" {{on "click" (fn this.updateStatus todo.id "completed"
+            )}}>Complete</button>
+        <button class="deleteBtn" type="button" {{on "click" (fn this.deleteTodo todo.id)}}>Delete</button>
```
Comment (hariharasuthan-balaji) [inline app/components/todos-list.hbs]: I see a lot of repeated code in this file. Lets optimize it.

**Review by hariharasuthan-balaji — CHANGES_REQUESTED** [review]

File: package.json [hunk 2]
```diff
@@ -36,6 +37,7 @@
     "ember-cli-dependency-checker": "^3.2.0",
     "ember-cli-htmlbars": "^5.7.1",
     "ember-cli-inject-live-reload": "^2.0.2",
+    "ember-cli-sass": "^11.0.1",
```
Comment (hariharasuthan-balaji) [inline package.json:39]: why do we need sass packages.

File: package.json [hunk 3]
```diff
@@ -59,7 +62,9 @@
     "npm-run-all": "^4.1.5",
     "prettier": "^2.3.0",
     "qunit": "^2.15.0",
-    "qunit-dom": "^1.6.0"
+    "qunit-dom": "^1.6.0",
+    "sass": "^1.80.6",
+    "sortablejs": "^1.15.3"
```
Comment (hariharasuthan-balaji) [inline package.json]: Do we need this ?

File: package.json [hunk 4]
```diff
@@ -46,6 +48,7 @@
     "ember-page-title": "^6.2.2",
     "ember-qunit": "^5.1.4",
     "ember-resolver": "^8.0.2",
+    "ember-sortablejs": "^2.0.0-beta.6",
```
Comment (hariharasuthan-balaji) [inline package.json]: required ?

File: package.json [hunk 5]
```diff
@@ -24,6 +24,7 @@
   },
   "devDependencies": {
     "@ember/optional-features": "^2.0.0",
+    "@ember/render-modifiers": "^2.1.0",
```
Comment (hariharasuthan-balaji) [inline package.json]: why do we need this ?

File: app/components/todo-item.hbs [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    class="todoTitleInput"
+    aria-label={{@todo.title}}
+    {{on "blur" (fn @onEditTaskTitle @todo.id)}}
+    value={{@todo.title}}
+  />
+  {{#each @statusActions as |action|}}
+    <button
+      class="{{action.buttonClass}}"
+      type="button"
+      {{on "click" (fn @updateStatus @todo.id action.status)}}
```
Comment (hariharasuthan-balaji) [inline app/components/todo-item.hbs:12]: Why do we need an `on` modifier ?

**Review by hariharasuthan-balaji — CHANGES_REQUESTED** [review]

File: package.json (hunk 4 above)
Comment (Pugazhenthiran-freshworks) [reply package.json]: No. Initially I tried to do draggable todo list. So, added at that time. Will remove

File: package.json (hunk 3 above)
Comment (Pugazhenthiran-freshworks) [reply package.json]: No. Initially I tried to do draggable todo list. So, added at that time. Will remove

File: package.json (hunk 2 above)
Comment (Pugazhenthiran-freshworks) [reply package.json:39]: I used .sccs. So added it

File: app/components/todo-item.hbs (hunk 6 above)
Comment (Pugazhenthiran-freshworks) [reply app/components/todo-item.hbs:12]: added on modifier to get click event from the button. Changing the status of the task using this on click event

File: package.json (hunk 5 above)
Comment (Pugazhenthiran-freshworks) [reply package.json]: Initially I used did-insert life cycle hook. I added for that. Will remove

File: app/components/todo-item.hbs (hunk 6 above)
Comment (hariharasuthan-balaji) [reply app/components/todo-item.hbs:12]: Please go through the inbuilt `action` helper and use it. Lets not add new packages and make the app heavy.

Comment (Pugazhenthiran-freshworks) [reply app/components/todo-item.hbs:12]: As we discussed, the "action" helper is not supported in 3.2.7. Also, "on" is inbuilt modifier.

**Review by hariharasuthan-balaji — APPROVED** [review]

**Pull request merged**